CryptoPkg/Library/BaseCryptLibMbedTls/BaseCryptLib.inf

```diff
@@ -57,9 +56,15 @@
   Rand/CryptRand.c
 
   SysCall/CrtWrapper.c
+  SysCall/TimerWrapper.c
+
+[Sources.IA32, Sources.X64, Sources.AARCH64, Sources.RISCV64, Sources.LOONGARCH64]

```

Comment:
Oh... Urgh. I hadn't fully realised the implications.

Look, I know you've already spend a silly amount of time untangling this mess. But as a frequent complainer about parts of the project that do "enumerate all architectures that aren't x86 and treat them as the exception" ... I don't really feel great about doing the same for Arm.

Could we split this up at the .inf and functional level instead? I.e. drop Arm support in this file and add a clone called BaseCryptLibNoSoftFloat.inf, just for Arm?

Reply:
NoHardFloat obvs...

Reply:
In that case, perhaps it would be better to create a version of OpensslLib, e.g., OpensslLibNoFp.inf, that incorporates only the code that can be used without floating point support.

Reply:
Wasn't the main reason for integrating mbedTls in the first place code size issues with openssl?

Reply:
Yes

Reply:
And wouldn't that be a problem for ARM targets especially?